apps/mobile/src/providers/onboarding-provider/onboarding-provider.tsx

```diff
@@ -0,0 +1,63 @@
+import { createContext, useMemo, useState } from 'react';
+
+import {
+  OnBoardingContextType,
+  UserInfo,
+  UpdateUserInfo,
+  UpdateUserLocation,
+} from './onboarding-provider.types';
+import { useAuth } from '../../hooks/use-auth-provider';
+
+const intialState: UserInfo = {
+  name: '',
+  password: '',
+  email: '',
+  location: {
+    latitude: 0,
+    longitude: 0,
+  },
+};
+
+export const OnBoardingContext = createContext<OnBoardingContextType>({} as OnBoardingContextType);
```

Comment:
I believe we could streamline this. Instead of using a context, WDYT about passing parameters through routes? https://reactnavigation.org/docs/params